lighthouse-core/audits/non-composited-animations.js

```diff
@@ -0,0 +1,145 @@
+    '[Learn more](https://developers.google.com/web/fundamentals/performance/rendering/stick-to-compositor-only-properties-and-manage-layer-count)',
+  /** [ICU Syntax] Label identifying the number of animations that are not composited. */
+  displayValue: `{itemCount, plural,
+  =1 {# animated element found}
+  other {# animated elements found}
+  }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/**
+ * Each failure reason is represented by a bit flag. The bit shift operator '<<' is used to define which bit corresponds to each failure reason.
+ * https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/animation/compositor_animations.h;l=58;drc=f959aba6099c5d767f885b7f4632955b463dc41d?originalUrl=https:%2F%2Fcs.chromium.org%2F
+ * @type {{flag: number, text: string}[]}
+ */
+const ACTIONABLE_FAILURE_REASONS = [
+  {
+    flag: 1 << 13,
+    text: 'Unsupported CSS Property',
+  },
+];
+
+/**
+ * Return list of actionable failure reasons and a boolean if some reasons are not actionable.
+ * Each flag is a number with a single bit set to 1 in the position corresponding to a failure reason.
+ * We can check if a specific bit is true in the failure coding using bitwise and '&' with the flag.
+ * @param {number} failureCode
+ * @return {string[]}
+ */
+function getActionableFailureReasons(failureCode) {
+  /** @type {string[]} */
+  const failureReasons = [];
+  ACTIONABLE_FAILURE_REASONS.forEach(reason => {
+    if (failureCode & reason.flag) {
+      failureReasons.push(reason.text);
+    }
+  });
+  return failureReasons;
+}
+
+class NonCompositedAnimations extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'non-composited-animations',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['TraceElements', 'HostUserAgent'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    // This audit reqiures m86
+    const match = artifacts.HostUserAgent.match(/Chrome\/(\d+)/);
+    if (!match || Number(match[1]) < 86) {
+      return {
+        score: 1,
+        notApplicable: true,
+      };
+    }
+
+    const animatedElements = artifacts.TraceElements
+      .filter(e => e.traceEventType === 'animation');
+
+    /** @type {LH.Audit.Details.TableItem[]} */
+    const results = [];
+    animatedElements.forEach(element => {
+      /** @type LH.Audit.Details.NodeValue */
+      const node = {
+        type: 'node',
+        path: element.devtoolsNodePath,
+        selector: element.selector,
+        nodeLabel: element.nodeLabel,
+        snippet: element.snippet,
+      };
+
+      /** @type LH.Audit.Details.TableItem[] */
+      const items = [];
```

Comment:
you could also do the filter/map combo here as well.

Reply:
Changed but see https://github.com/GoogleChrome/lighthouse/pull/11105#discussion_r463714102